src/tests/activities.test.js

```diff
@@ -1,3 +1,130 @@
-test('should pass a basic test', () => {
-  expect(1 + 1).toBe(2);
+import request from 'supertest';
+import app from '../app';
+import activity from '../models/Activity';
+import activitySeed from '../seeds/activities';
+
+const activities = {
+  name: 'voley',
+  description: 'basic training that consists of throwing soft punches into the air with ertert erte reter dfd er ert  ert',
+  isActive: true,
+};
+
+const ActivityBad = {
+  name: '',
+  description: '',
+  isActive: true,
+};
+
+const updatedActivity = {
+  name: 'basquetball',
+  description: 'This is the most practiced sport nowadays by teams of men and women, both professionally and amateurishly.',
+  isActive: false,
+};
+
+beforeAll(async () => {
+  await activity.collection.insertMany(activitySeed);
+});
+
+describe('POST /api/activities', () => {
+  test('should create a activity', async () => {
+    const res = await request(app).post('/api/activities').send(activities);
+    expect(res.status).toBe(201);
+    expect(res.body).toBeDefined();
+    expect(res.error).toBeFalsy();
+  });
+
+  test('should return a error', async () => {
+    const res = await request(app).post('/api/activities').send(ActivityBad);
+    expect(res.status).toBe(400);
+    expect(res.body.data).toBeUndefined();
+    expect(res.error).toBeTruthy();
+  });
+});
+
+describe('PUT /api/activities/:id', () => {
+  test('should update a activity', async () => {
+    const res = await request(app).put('/api/activities/6466b3e7a21fd4069bcaf7c0').send(updatedActivity);
+    expect(res.status).toBe(200);
+    expect(res.body).toBeDefined();
+    expect(res.error).toBeFalsy();
+  });
+
+  test('should return a error', async () => {
+    const res = await request(app).put('/api/activities/6466b3e7a21fd4069bcaf7c0').send(ActivityBad);
+    expect(res.status).toBe(400);
+    expect(res.body.data).toBeUndefined();
+    expect(res.error).toBeTruthy();
+  });
+
+  test('should return a error, id not valid', async () => {
+    const res = await request(app).put('/api/activities/6462261').send(updatedActivity);
```

Comment:
You can also add: expect(res.body.data).toBeUndefined();

Reply:
> You can also add: expect(res.body.data).toBeUndefined();

In this test i don't use "toBeUndefined" because the controller "put" returns an object, it has not property "data" and does not return anything undefined, i don't know if that it should be like that but it comes from the controller